File: fabric/src/main/java/net/shiroha233/roadweaver/network/fabric/ClientNetBridgeImpl.java
package net.shiroha233.roadweaver.network.fabric;

import net.minecraft.resources.ResourceLocation;
import net.shiroha233.roadweaver.client.map.MapLoadPhase;

/**
 * Fabric 平台网络桥接实现
 */
public final class ClientNetBridgeImpl {
    private ClientNetBridgeImpl() {}

    public static void requestSnapshot(int requestSeq,
                                       ResourceLocation dimensionId,
                                       MapLoadPhase phase,
                                       int responseIndex,
                                       int minX,
                                       int minZ,
                                       int maxX,
                                       int maxZ) {
        MapNetworkFabric.requestSnapshot(requestSeq, dimensionId, phase, responseIndex, minX, minZ, maxX, maxZ);
    }

    public static void requestTeleport(int x, int y, int z) {
        MapNetworkFabric.requestTeleport(x, y, z);
    }

    public static void requestManualConnect(int ax, int az, int bx, int bz) {
        MapNetworkFabric.requestManualConnect(ax, az, bx, bz);
    }

    public static void requestSearch(int requestSeq, ResourceLocation dimensionId, String query) {
        MapNetworkFabric.requestSearch(requestSeq, dimensionId, query);
    }
}

File: fabric/src/main/kotlin/net/shiroha233/roadweaver/network/fabric/ClientNetBridgeImpl.kt
package net.shiroha233.roadweaver.network.fabric

object ClientNetBridgeImpl {
    @JvmStatic
    fun requestSnapshot(minX: Int, minZ: Int, maxX: Int, maxZ: Int) {
        MapNetworkFabric.requestSnapshot(minX, minZ, maxX, maxZ)
    }

    @JvmStatic
    fun requestTeleport(x: Int, y: Int, z: Int) {
        MapNetworkFabric.requestTeleport(x, y, z)
    }

    @JvmStatic
    fun requestManualConnect(ax: Int, az: Int, bx: Int, bz: Int) {
        MapNetworkFabric.requestManualConnect(ax, az, bx, bz)
    }
}
